webknossos/webknossos/annotation/annotation.py

```diff
@@ -434,6 +436,52 @@ def _write_to_zip(self, zipfile: ZipFile) -> None:
                 layer_content,
             )
 
+    def get_annotation_dataset(self) -> Dataset:
+        """Returns a streamed dataset, incorporating fallback layers and mappings."""
+        from webknossos.client.context import _get_context
+
+        if self.annotation_id is None:
+            raise ValueError(
+                "The annotation_id is not set, cannot get the corresponding dataset."
+            )
+
+        context = _get_context()
+        token: Optional[str]
+        if self.organization_id is None:
+            token = context.required_token
+            organization_id = context.organization_id
+        else:
+            token = context.token
+            organization_id = self.organization_id
+            # avoid requiring authentication
+            if token is not None:
+                if organization_id != context.organization_id:
+                    warnings.warn(
+                        "The annotation used in with get_annotation_dataset "
+                        + "specifies a different organization id than the current context. "
+                        + f"The annotation uses {organization_id}, the context {context.organization_id}.",
+                        RuntimeWarning,
+                    )
+
+        dataset_info_response = dataset_info.sync_detailed(
+            organization_name=organization_id,
+            data_set_name=self.dataset_name,
+            client=context.generated_client,
+        )
+        assert dataset_info_response.status_code == 200, dataset_info_response
+        parsed = dataset_info_response.parsed
+        assert parsed is not None
+
+        # token = sharing_token or wk_context.datastore_token
```

Comment:
can this be removed? would datastore tokens be necessary in certain scenarios? @fm3 can you answer this? I'm asking since the code currently does not deal with datastore tokens (at least, I don't see it).

Reply:
All datastore routes now also take the user token, if sent via X-Auth-Token or as get param. The expiring datastore tokens still work as well (and can be sent in the same way) but I discussed with @jstriebel that it’s fine if the python client uses the non-expiring user token, since the user will probably need to insert that into their python code anyway and should be aware of the required secrecy.

Reply:
ok, then I'll remove that line.